tools/protodoc/protodoc.py

```diff
@@ -691,6 +720,20 @@ def visit_file(self, file_proto, type_context, services, msgs, enums):
         if all(len(msg) == 0 for msg in msgs) and all(len(enum) == 0 for enum in enums):
             has_messages = False
 
+        v2_link = ""
+        if file_proto.name in self.v2_mapping:
+            # TODO(phlax): remove _v2_ from filepath once sed mangling is removed
+            v2_filepath = f"envoy_v2_api_file_{self.v2_mapping[file_proto.name]}"
+            v2_text = v2_filepath.split('/', 1)[1]
+            v2_url = f"v{ENVOY_LAST_V2_VERSION}:{v2_filepath}"
+            v2_link = V2_LINK_TEMPLATE.render(v2_url=v2_url, v2_text=v2_text)
+
+            # TODO(phlax): move this to a separate bazel task
+            with open(REDIRECTS_FILE, "a") as f:
```

Comment:
Is this code safe? WE have multiple protodoc instances running in parallel potentially, all writing to the same file?

Reply:
hmm - really good point - most probably its not safe

ill rethink this - either shift to a separate target or just hardcode the redirects file